frontend/src/metabase/admin/settings/app/components/SettingsEditor/SettingsEditor.jsx

```diff
@@ -39,7 +39,7 @@ import { SettingsSection } from "./SettingsSection";
 
 const mapStateToProps = (state, props) => {
   return {
-    settings: getSettings(state, props),
+    settingsByKey: getSettingsByKey(state, props),
```

Comment:
Everywhere we use `settings` (array) we always either convert it to an object, or just try to find the desired setting, so passing `settingsByKey` makes those tasks more simple.

Reply:
I understand why you made this change and It makes a lot of sense, but I would really prefer that you not make fundamental changes to the settings code like this in a PR that is just trying to add a few new settings. This complicates the code and introduces an inconsistent pattern than we're going to eliminate in a different and more wholistic way as part of the larger settings refactor.

Reply:
Can you please not make fundamental changes to how all settings components work in an unrelated PR 🙏

Reply:
that's fair, I'll revert these settings changes.